### PR TITLE
feat: add new round number fn

### PR DESCRIPTION
**Pull request by nvelazco**

### CODE REVIEW

File: src/round-number.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * roundNumber(0, 2.8) // returns 3
+ * @param {Number} decimals number of decimals to round to
+ * @param {Number} value number to be rounded
+ * @returns {Number} rounded number
+ */
+function roundNumber(precision, value) {
+  return applyTo(
+    value,
+    compose(
+      // Round to two decimal places
```
Comment (gaston-flores) [inline src/round-number.js:21]: Only two? It looks like this is a parameter 😛

File: src/round-number.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+'use strict';
+const roundNumber = require('./round-number');
+
+test('should return the number rounder to 0 decimals, rounding up when has to', () => {
```
Comment (gaston-flores) [inline src/round-number.test.js:4]:
```suggestion
test('should return the number rounded to 0 decimals, rounding up when it has to', () => {
```

File: src/round-number.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      total =>
+        Number.isFinite(total) ? +(Math.round(total + `e+${precision}`) + `e-${precision}`) : total,
+      // Coerce values higher than `Number.MAX_SAFE_INTEGER` to `Infinity`
+      when(lt(MAX_MATH_DELTA), always(Infinity)),
+      // Coerce values lower than `0.0000001` to `0`
+      when(gt(MIN_MATH_DELTA), always(0))
```
Comment (gaston-flores) [inline src/round-number.js:27]: What will happen to negative numbers?

Comment (nvelazco) [reply src/round-number.js]: solved in the last commit <3

File: src/utils/_math-constants.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+/**
+ * The minimum value beneath which underflow to zero occurs.
+ * @type {Number}
+ */
+const MIN_MATH_DELTA = 1e-7;
```
Comment (nfantone) [inline src/utils/_math-constants.js:7]: Why are these constants needed in a separate module, exactly? Do we require them somewhere else?

File: src/round-number.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const { when, always, compose, lt, applyTo, gt, both } = require('ramda');
+const _curry2 = require('./utils/_curry2');
+const { MAX_MATH_DELTA, MIN_MATH_DELTA, MIN_NEG_MATH_DELTA } = require('./utils/_math-constants');
+
+/**
+ * Rounds a number to the next largest number for positives and the next smallest for negatives
```
Comment (nfantone) [inline src/round-number.js:8]:
```suggestion
 * Rounds a given number to the next largest number for positives and the next smallest for negatives.
```

Small wording change.

File: src/round-number.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * @example
+ * roundNumber(2, 2.151) // returns 2.15
+ * roundNumber(1, -2.63) // returns -2.7
+ * roundNumber(0, 2.8) // returns 3
+ * @param {Number} decimals number of decimals to round to
+ * @param {Number} value number to be rounded
+ * @returns {Number} rounded number
+ */
+function roundNumber(precision, value) {
```
Comment (nfantone) [inline src/round-number.js:17]: I would honestly call this just `round` - following `Math.round`.

File: src/round-number.test.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+const roundNumber = require('./round-number');
+
+test('should return the number rounded to 0 decimals, rounding up when has to', () => {
```
Comment (nfantone) [inline src/round-number.test.js:4]:
```suggestion
test('should return the number rounded to 0 decimals, rounding up when it has to', () => {
```

File: src/round-number.test.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
+'use strict';
```
Comment (nfantone) [inline src/round-number.test.js:1]: Would like to see at least one test on coercing non-number values.

e.g.:

```js
roundNumber('2.8')
```

File: src/utils/_math-constants.js (hunk 4 above)
Comment (nvelazco) [reply src/utils/_math-constants.js:7]: Exactly, these constants are used in `round-number` and `sum-prop` functions.

**Pull request merged**